### DISCUSSION TITLE
Having trouble using Payload CMS 3 with Next.js middleware (Node Error)

### DISCUSSION THREAD

**Discussion by m7medVision** [payloadcms/payload#6506] [Q&A] [answered]
Hi everyone,

I'm trying to use payload cms 3.0 with `middleware.ts` to do check user auth, but I'm encountering a Node error during the build process. Here's the error message:
```
Module build failed: UnhandledSchemeError: Reading from "node:os" is not handled by plugins (Unhandled scheme).
Webpack supports "data:" and "file:" URIs by default.
You may need an additional plugin to handle "node:" URIs.
```
I'm not very familiar with this error. Has anyone else faced this when using Payload CMS with middleware in Next.js? Any guidance on how to fix it or what plugin might be needed would be greatly appreciated!

**Comment by andershermansen** [comment, marked as answer]
Ref: https://nextjs.org/docs/app/building-your-application/routing/middleware#runtime

Middleware currently only supports the [Edge runtime](https://nextjs.org/docs/app/building-your-application/rendering/edge-and-nodejs-runtimes). The Node.js runtime can not be used.

I think you should move your auth check into a higher level layout instead.

**Reply by m7medVision** [reply]
Thank you, Mr. andershermansen

**Comment by malitov** [comment]
@m7medVision Middleware supports Node.js runtime now from Next.js v15.5
https://nextjs.org/docs/app/api-reference/file-conventions/middleware#runtime
